components/Cards/DetailsCard.js

```diff
@@ -322,6 +349,15 @@ export default function DetailsCard({
           />
         )}
         <StyledCardBody>
+          <StyledFormGroup>
+            {isEditingPlaceId ? (
+              <UploadImage
+                isImage={isImage}
+                setIsImage={setIsImage}
+                setPublicId={setPublicId}
+              />
+            ) : null}
+          </StyledFormGroup>
```

Comment:
Your solution is concise and functional, but there is a suggestion to adjust the conditional rendering. 

⚠️ If `setIsImage` and `setPublicId` are state setter functions (to be honest, this can't be reproduced in line 355), it's better to encapsulate them within handler functions before passing to components. This approach prevents direct modification of state from child components and allows for additional logic before state updates, enhancing maintainability and readability.

```suggestion
// Utilizing conditional rendering directly within the JSX to make the code cleaner
<StyledFormGroup>
  {
    // Only render the UploadImage component when isEditingPlaceId is true ✨
    isEditingPlaceId && (
      <UploadImage
        isImage={isImage} 
        setIsImage={setIsImage} 
        setPublicId={setPublicId} /
      />
    )
  }
</StyledFormGroup>
```